vm/systemSmartContracts/staking.go

```diff
@@ -1380,30 +1380,33 @@ func (s *stakingSC) getWaitingListRegisterNonceAndRewardAddress(args *vmcommon.C
 	return vmcommon.Ok
 }
 
-func (s *stakingSC) setOwner(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+func (s *stakingSC) setOwnersOnAddresses(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
 	if !s.flagStakingV2.IsSet() {
 		s.eei.AddReturnMessage("invalid method to call")
 		return vmcommon.UserError
 	}
 	if !bytes.Equal(args.CallerAddr, s.stakeAccessAddr) {
-		s.eei.AddReturnMessage("setOwner function not allowed to be called by address " + string(args.CallerAddr))
+		s.eei.AddReturnMessage("setOwnersOnAddresses function not allowed to be called by address " + string(args.CallerAddr))
 		return vmcommon.UserError
 	}
-	if len(args.Arguments) < 2 {
-		s.eei.AddReturnMessage(fmt.Sprintf("invalid number of arguments: expected min %d, got %d", 2, len(args.Arguments)))
-		return vmcommon.UserError
-	}
-	stakedData, err := s.getOrCreateRegisteredData(args.Arguments[0])
-	if err != nil {
-		s.eei.AddReturnMessage(err.Error())
+	if len(args.Arguments)%2 != 0 {
+		s.eei.AddReturnMessage("invalid number of arguments: expected an even number of arguments")
 		return vmcommon.UserError
 	}
-
-	stakedData.OwnerAddress = args.Arguments[1]
-	err = s.saveStakingData(args.Arguments[0], stakedData)
-	if err != nil {
-		s.eei.AddReturnMessage(err.Error())
-		return vmcommon.UserError
+	for i := 0; i < len(args.Arguments); i += 2 {
```

Comment:
i<len() -1 ?

Reply:
no need, there is a check that the len of arguments is an even number. So, supposing the len is 2, the loop instructions will be executed only once as the i == 0 (satisfy < 2 condition). After that iteration i becomes 2 and will no longer satisfy the condition i < 2 as 2 < 2 is false. Another example, suppose that len is 8. It will iterate exactly 4 times for i == 0, i == 2, i == 4 and i == 6. We have 3 tests (1 unit test and 2 semi-integrated tests) that check this iteration is done correctly,